net/socket.h

```diff
@@ -208,11 +345,31 @@ namespace net
 }
 
 namespace std {
+
 template<>
 struct hash<photon::net::EndPoint> {
-    hash<uint64_t> hasher;
     size_t operator()(const photon::net::EndPoint& x) const {
-        return hasher((x.addr.to_nl() << 16) | x.port);
+        if (x.is_ipv4()) {
+            hash<uint64_t> hasher;
+            return hasher((x.addr.to_nl() << 16) | x.port);
+        } else {
+            hash<std::string_view> hasher;
+            return hasher(std::string_view((const char*) &x, sizeof(x)));
```

Comment:
> do we need to distinguish between v4 and v6?

Because string_view hash is not balanced enough for integers. For simplicity, we call use string_view hash. 

https://xxhash.com/

Reply:
Will change both of them to string_view hash.